README.md

```diff
@@ -1,39 +1,411 @@
+Loggerator::Metrics.name        = 'myapp'
+```
+
+### Log Helpers
+
+#### `log`
+
+Use `log` to log any key-value pairs, they can be any value.
+
+Logs are sent to `stdout` by default
+
+```ruby
+log test1: "first", test2: 123  #=> app=myapp test1=first test2=123
+```
+
+Use blocks to emit timing information, assuming `test_method` does the following:
+```
+def test_method
+  log(sleep: 0.4)
+  sleep(0.4)
+end
+```
+
+Then you can see actual timing with:
+
+```ruby
+log timing: "test_method" do
+  test_method
+  log success: true
+end
+#=> app=myapp timing=test_method at=start
+#=> app=myapp sleep: 0.400
+#=> app=myapp success
+#=> app=myapp timing=test_method at=finish elapsed=0.401
+```
+
+#### `log_context`
+
+You can use `log_context` to add values to all encompassed `log` commands
+
+```ruby
+log_context group: "test_method" do
+  log try: 1
+  test_method
+  log success: true
+end
+#=> app=myapp group=test_method try=1
+#=> app=myapp group=test_method sleep: 0.400
+#=> app=myapp group=test_method success
+```
+
+#### `log_error`
+
+Log exception information using `log_error`.  It will log the class, message, and backtrace.  Every frame of the backtrace will be logged on its own line and is reverse-ordered so that the most relevant stack frames are last. All the logs will be tied together by a common `exception_id`.
+
+Errors logs are sent to `stderr` by default.
+
+```ruby
+def emit_error
+  raise "this is an error"
+rescue
+  log_error $!
+end
+
+emit_error
+#=> app=myapp exception_id=70113140892880 backtrace="(pry):64:in `<main>'"
+#=> app=myapp exception_id=70113140892880 backtrace="(pry):53:in `emit_error'"
+#=> app=myapp exception_id=70113140892880 exception class=RuntimeError message="this is an error"
+```
+
+You can add additional key-value pairs to `log_error`
+
+```ruby
+log_error $!, try: 2
+#=> app=myapp exception_id=70113140892880 exception class=RuntimeError message="this is an error" try=2
+```
+
+### Formatting Rules
+
+#### Strings
+
+Loggerator will handle strings sanely by wrapping in quotes when needed:
+
+```ruby
+log no_spaces: "first", spaces: "second test"
+#=> app=myapp no_spaces=first spaces="second test"
+```
+
+And handles embedded quotes:
+```ruby
+log double: %q[test with a "double-quote"], single: %q[test with a 'single-quote']
+#=> app=myapp double='test with a "double-quote"' single="test with a 'single-quote'"
+
+log embedded_quotes: %q[test with a "double-quote with a 'single-quote' inside"]
+#=> app=myapp embedded_quotes="test with a \"double-quote with a 'single-quote' inside\""
+```
+
+#### Numbers
+
+Floats will be formatted to three digits of precision, all other numbers will be shown as-is.
+
+```ruby
+log pi: 3.14159, integer: 123456789  #=> app=myapp pi=3.142 integer=123456789"
+```
+
+#### Booleans
+
+True values will only show the key, false values are shown as is.
+
+```ruby
+log on: true, off: false  #=> app=myapp on off=false"
+```
+
+#### Nil
+
+Nil values will be removed
```

Comment:
This is a hold-over from Pliny.  I think its useful in that `nil` values don't clog up the logs. 
I don't think the intent was to remove items from the log_context but that would be a side-effect.